### PR TITLE
PLATUI-1192: Deprecating old route whilst persisting value from REFER…

### PR DESCRIPTION
**Pull request by JoPintoPaul**
…ER header

### CODE REVIEW

File: app/controllers/ReportProblemController.scala [hunk 1]
```diff
@@ -130,48 +130,65 @@ class ReportProblemController @Inject() (
...
 
-  def index(service: Option[String]) = Action { implicit request =>
-    Ok(page(ReportProblemFormBind.emptyForm(service = service), service))
+  def index(service: Option[String], referrerUrl: Option[String]) = Action { implicit request =>
+    val referrer = referrerUrl.orElse(request.headers.get(REFERER))
+    Ok(page(ReportProblemFormBind.emptyForm(service, referrer), service, referrerUrl))
+  }
+
+  def indexDeprecated(service: Option[String]) = Action { implicit request =>
+    val referrer = request.headers.get(REFERER)
```
Comment (matthewmascord) [inline app/controllers/ReportProblemController.scala:139]: shouldn't we be getting the referrerUrl on the deprecated route too?

File: conf/contact.routes [hunk 2]
```diff
@@ -26,11 +26,11 @@ GET         /contact-hmrc-unauthenticated                @controllers.ContactHmr
...
 GET         /contact-hmrc/thanks-unauthenticated         @controllers.ContactHmrcController.thanks
 
-GET         /report-technical-problem                    @controllers.ReportProblemController.index(service: Option[String])
-POST        /report-technical-problem                    @controllers.ReportProblemController.submit(service: Option[String])
+GET         /report-technical-problem                    @controllers.ReportProblemController.index(service: Option[String], referrerUrl: Option[String])
+POST        /report-technical-problem                    @controllers.ReportProblemController.submit(service: Option[String], referrerUrl: Option[String])
 GET         /report-technical-problem/thanks             @controllers.ReportProblemController.thanks
-GET         /problem_reports_nonjs                       @controllers.ReportProblemController.index(service: Option[String])
-POST        /problem_reports_nonjs                       @controllers.ReportProblemController.submit(service: Option[String])
+GET         /problem_reports_nonjs                       @controllers.ReportProblemController.indexDeprecated(service: Option[String])
```
Comment (matthewmascord) [inline conf/contact.routes:32]: Add referrerUrl as Option[String] to deprecated URL?

File: app/controllers/ReportProblemController.scala (hunk 1 above)
Comment (JoPintoPaul) [reply app/controllers/ReportProblemController.scala]: Mea culpa - hadn't included that commit in the PR, pushed up now.

File: conf/contact.routes (hunk 2 above)
Comment (JoPintoPaul) [reply conf/contact.routes]: Mea culpa - hadn't included that commit in the PR, pushed up now.

**Pull request merged**